File: python/examples/accounts/regions/v1/list_regions_sample.py
# -*- coding: utf-8 -*-
# Copyright 2024 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""A module to list all the Regions for a given Merchant Center account."""

# [START merchantapi_list_regions]
from examples.authentication import configuration
from examples.authentication import generate_user_credentials
from google.shopping.merchant_accounts_v1 import ListRegionsRequest
from google.shopping.merchant_accounts_v1 import RegionsServiceClient

_ACCOUNT = configuration.Configuration().read_merchant_info()


def get_parent(account_id):
  return f"accounts/{account_id}"


def list_regions():
  """Lists all the regions for a given Merchant Center account."""

  # Gets OAuth Credentials.
  credentials = generate_user_credentials.main()

  # Creates a client.
  client = RegionsServiceClient(credentials=credentials)

  # Creates parent to identify the account from which to list all regions.
  parent = get_parent(_ACCOUNT)

  # Creates the request.
  request = ListRegionsRequest(parent=parent)

  # Makes the request and catches and prints any error messages.
  try:
    response = client.list_regions(request=request)
    count = 0
    print("Sending list regions request:")
    for element in response:
      print(element)
      count += 1
    print(f"The following count of elements were returned: {count}")

  except RuntimeError as e:
    print("List region request failed!")
    print(e)


if __name__ == "__main__":
  list_regions()

# [END merchantapi_list_regions]

File: php/examples/accounts/regions/v1/ListRegionsSample.php
<?php
/**
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
require_once __DIR__ . '/../../../../vendor/autoload.php';
require_once __DIR__ . '/../../../Authentication/Authentication.php';
require_once __DIR__ . '/../../../Authentication/Config.php';
// [START merchantapi_list_regions]
use Google\ApiCore\ApiException;
use Google\Shopping\Merchant\Accounts\V1\Client\RegionsServiceClient;
use Google\Shopping\Merchant\Accounts\V1\ListRegionsRequest;

/**
 * This class demonstrates how to list all the regions for a given Merchant Center account.
 */
class ListRegionsSample
{

    private static function getParent(string $accountId): string
    {
        return sprintf("accounts/%s", $accountId);
    }

    public static function listRegionsSample(array $config): void
    {
        // Gets the OAuth credentials to make the request.
        $credentials = Authentication::useServiceAccountOrTokenFile();

        // Creates options config containing credentials for the client to use.
        $options = ['credentials' => $credentials];

        // Creates a client.
        $regionsServiceClient = new RegionsServiceClient($options);

        // Creates parent to identify the account from which to list all regions.
        $parent = self::getParent($config['accountId']);

        try {
            $request = new ListRegionsRequest([
                'parent' => $parent
            ]);

            print "Sending list regions request:\n";
            $response = $regionsServiceClient->listRegions($request);

            $count = 0;
            foreach ($response->iterateAllElements() as $element) {
                print $element->getName() . PHP_EOL;
                $count++;
            }
            print "The following count of elements were returned: ";
            print $count . PHP_EOL;

        } catch (ApiException $e) {
            print $e->getMessage();
        }
    }

    public function callSample(): void
    {
        $config = Config::generateConfig();
        self::listRegionsSample($config);
    }
}

$sample = new ListRegionsSample();
$sample->callSample();
// [END merchantapi_list_regions]
